lexicons/com/atproto/sync/listReposByCollection.json

```diff
@@ -0,0 +1,45 @@
+{
+  "lexicon": 1,
+  "id": "com.atproto.sync.listReposByCollection",
+  "defs": {
+    "main": {
+      "type": "query",
+      "description": "Enumerates all the DIDs which have records with the given collection NSID.",
+      "parameters": {
+        "type": "params",
+        "required": ["collection"],
+        "properties": {
+          "collection": { "type": "string", "format": "nsid" },
+          "limit": {
+            "type": "integer",
+            "minimum": 1,
+            "maximum": 500,
+            "default": 50
```

Comment:
I think the default should be much higher, paging through millions at 50/call would be bad. The db access is super cheap, this is just about what a reasonable network blob is. 32 byte did * 1000 would only be 32kB of data so at least that. Maybe default 500 or 1000, upper limit 5000 or 10000.

Reply:
I think a common case is going to be folks just iterating through hundreds of accounts, not millions, right? eg, looking for all feed generator declarations, or statusphere users. I think folks doing this bulk should configure the limit; it is nice to have it smaller for devs poking around with `curl`.

I'll bump to default=100 and max=5000

(I don't feel super strongly about this and could bump higher as well)

Reply:
My inclination would be to follow `sync.listRepos` which has a pretty similar use case.  The number of repos on a host vs. within a collection seems like they both span large and small lists.

Reply:
`sync.listRepos` has min=1, default=500, max=1000.

I went with min=1, default=500, max=2000 to be mostly-aligned